http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoPubsubMessageSubscriber.scala

```diff
@@ -534,9 +556,185 @@ class LeoPubsubMessageSubscriber[F[_]: Timer: ContextShift](
+        _ <- clusterResult.traverse(result => gkeInterp.pollCluster(result)).adaptError {
+          case e =>
+            PubsubKubernetesError(
+              KubernetesError(e.getMessage(), ctx.now, ErrorAction.CreateGalaxyApp, ErrorSource.Cluster, None),
+              msg.appId,
+              false,
+              msg.cluster.map(_.nodepoolId),
+              msg.cluster.map(_.clusterId)
+            )
+        }
+        _ <- gkeInterp.createAndPollNodepool(msg.nodepoolId, msg.appId).adaptError {
+          case e =>
+            PubsubKubernetesError(
+              KubernetesError(e.getMessage(), ctx.now, ErrorAction.CreateGalaxyApp, ErrorSource.Nodepool, None),
+              msg.appId,
+              false,
+              Some(msg.nodepoolId),
+              None
+            )
+        }
+      } yield ()
 
-  def handleDeleteAppMessage(msg: DeleteAppMessage): F[Unit] = F.unit
+      diskCreation = createDiskForApp(msg).adaptError {
+        case e =>
+          PubsubKubernetesError(
+            KubernetesError(e.getMessage(), ctx.now, ErrorAction.CreateGalaxyApp, ErrorSource.Disk, None),
+            msg.appId,
+            false,
+            None,
+            None
+          )
+      }
+      parPreAppCreationSetup = List(createClusterAndNodepool, diskCreation).parSequence.flatMap(_ => F.unit)
+
+      task = for {
+        _ <- parPreAppCreationSetup
+        _ <- gkeInterp.createAndPollApp(msg).adaptError {
+          case e =>
+            PubsubKubernetesError(
+              KubernetesError(e.getMessage(), ctx.now, ErrorAction.CreateGalaxyApp, ErrorSource.App, None),
+              msg.appId,
+              false,
+              None,
+              None
+            )
+        }
+      } yield ()
+
+      _ <- asyncTasks.enqueue1(
+        Task(ctx.traceId, task, Some(handleKubernetesError), ctx.now)
+      )
+    } yield ()
+
+  private def handleKubernetesError(e: Throwable): F[Unit] =
+    e match {
+      case e: PubsubKubernetesError =>
+        for {
+          _ <- appErrorQuery.save(e.appId, e.dbError).transaction
+          _ <- appQuery.updateStatus(e.appId, AppStatus.Error).transaction
+          _ <- e.clusterId.traverse(clusterId =>
+            kubernetesClusterQuery.updateStatus(clusterId, KubernetesClusterStatus.Error).transaction
+          )
+          _ <- e.nodepoolId.traverse(nodepoolId =>
+            nodepoolQuery.updateStatus(nodepoolId, NodepoolStatus.Error).transaction
+          )
+          _ <- logger.error(e)(s"updating db state for an async error for app ${e.appId}")
+        } yield ()
+      case _ =>
+        F.raiseError(new RuntimeException("handleKubernetesError should not be used with a non kubernetes error"))
+    }
+
+  private def deleteDiskForApp(diskId: DiskId)(
+    implicit ev: ApplicativeAsk[F, AppContext]
+  ): F[Unit] =
+    for {
+      ctx <- ev.ask
+      _ <- handleDeleteDiskMessage(DeleteDiskMessage(diskId, Some(ctx.traceId)), true)
```

Comment:
minor, but I would probably leave `handleDeleteDiskMessage` untouched, and create another function that takes a `sync` param, something like `private def deleteDisk(diskId: DiskId])(implicit ev: ApplicativeAsk[F, AppContext])` because I see `handleXXX` functions as top level function only called in `messageResponder`, and wouldn't expect it being used by other functions.

Reply:
I was a little hesitant, but for now the desired functionality is identical and I like not duplicating the code. There are unit tests to cover both success and failure for disks, so if they diverge and it breaks for kubernetes, we can evaluate if we want to split them

Reply:
I don't mean to duplicate code...but mostly just to to break out body of `handleDeleteDiskMessage` to another function that can be shared